src/net.c

```diff
@@ -295,97 +295,66 @@ coap_pop_next(coap_context_t *context) {
 }
 
 #if COAP_CLIENT_SUPPORT
-static size_t
-coap_get_session_client_psk(
-  const coap_session_t *session,
-  const uint8_t *hint, size_t hint_len,
-  uint8_t *identity, size_t *identity_len, size_t max_identity_len,
-  uint8_t *psk, size_t max_psk_len
+const coap_bin_const_t *
+coap_get_session_client_psk_key(
+  const coap_session_t *session
 ) {
-  const coap_dtls_cpsk_info_t *psk_info;
-  (void)hint;
-  (void)hint_len;
-
-  if (session->psk_identity && session->psk_key) {
-    if (session->psk_identity->length <= max_identity_len &&
-        session->psk_key->length <= max_psk_len) {
-      memcpy(identity, session->psk_identity->s, session->psk_identity->length);
-      memcpy(psk, session->psk_key->s, session->psk_key->length);
-      *identity_len = session->psk_identity->length;
-      return session->psk_key->length;
-    }
-  }
-  psk_info = &session->cpsk_setup_data.psk_info;
-  if (psk_info->identity.s && psk_info->identity.length > 0 &&
-      psk_info->key.s && psk_info->key.length > 0) {
-    if (psk_info->identity.length <= max_identity_len &&
-      psk_info->key.length <= max_psk_len) {
-      memcpy(identity, psk_info->identity.s, psk_info->identity.length);
-      memcpy(psk, psk_info->key.s, psk_info->key.length);
-      *identity_len = psk_info->identity.length;
-      return psk_info->key.length;
-    }
+
+  if (session->psk_key) {
+    return session->psk_key;
   }
+  if (session->cpsk_setup_data.psk_info.key.length)
+    return &session->cpsk_setup_data.psk_info.key;
+
   /* Not defined in coap_new_client_session_psk2() */
-  *identity_len = 0;
-  return 0;
+  return NULL;
 }
 #endif /* COAP_CLIENT_SUPPORT */
 
+const coap_bin_const_t *
+coap_get_session_client_psk_identity(
+  const coap_session_t *session
```

Comment:
At some point, I would like new and existing code to comply with the coding styles described in [Section 5.2 of the CONTRIBUTION guidelines](https://github.com/obgm/libcoap/blob/5d9b29ce656f55368a63b04ef2030f38fda8e17c/CONTRIBUTE#L129).

Reply:
Sorry - result of cut'n paste.  Re-aligned this function and several after it. Code pushed.